srml/scored-pool/src/lib.rs

```diff
@@ -0,0 +1,373 @@
+		pub fn issue_candidacy(origin) {
+			let who = ensure_signed(origin)?;
+			ensure!(Self::find_in_pool(&who).is_err(), "already a member");
+
+			T::Currency::reserve(&who, T::CandidateDeposit::get())
+				.map_err(|_| "balance too low")?;
+
+			let mut pool = <Pool<T, I>>::get();
+			// can be inserted as last element in pool, since entities with
+			// `None` are always sorted to the end.
+			pool.push((who.clone(), None));
+			<Pool<T, I>>::put(&pool);
+
+			Self::deposit_event(RawEvent::CandidateAdded);
+		}
+
+		/// An entity withdraws candidacy and gets its deposit back.
+		///
+		/// If the entity is part of the `Members`, then the highest member
+		/// of the `Pool` that is not currently in `Members` is immediately
+		/// placed in the set instead.
+		pub fn withdraw_candidacy(origin) {
+			let who = ensure_signed(origin)?;
+
+			Self::remove_member(who)?;
+			Self::deposit_event(RawEvent::CandidateWithdrew);
+		}
+
+		/// Kick a member `who` from the set.
+		///
+		/// May only be called from `KickOrigin` or root.
+		pub fn kick(
+			origin,
+			dest: <T::Lookup as StaticLookup>::Source
+		) {
+			T::KickOrigin::try_origin(origin)
+				.map(|_| ())
+				.or_else(ensure_root)
+				.map_err(|_| "bad origin")?;
+
+			let who = T::Lookup::lookup(dest)?;
+			Self::remove_member(who)?;
+			Self::deposit_event(RawEvent::CandidateKicked);
+		}
+
+		/// Score a member `who` with `score`.
+		///
+		/// May only be called from `ScoreOrigin` or root.
+		pub fn score(
+			origin,
+			dest: <T::Lookup as StaticLookup>::Source,
+			score: T::Score
+		) {
+			T::ScoreOrigin::try_origin(origin)
+				.map(|_| ())
+				.or_else(ensure_root)
+				.map_err(|_| "bad origin")?;
+
+			let who = T::Lookup::lookup(dest)?;
+			let mut pool = <Pool<T, I>>::get();
+			let location = Self::find_in_pool(&who)?;
+			pool.remove(location);
+
+			// we binary search the pool (which is sorted descending by score).
+			// if there is already an element with `score`, we insert
+			// right before that. if not, the search returns a location
+			// where we can insert while maintaining order.
+			let item = (who.clone(), Some(score.clone()));
+			let location = pool
+				.binary_search_by_key(
+					&Reverse(score),
+					|(_, maybe_score)| Reverse(maybe_score.unwrap_or_default())
+				)
+				.unwrap_or_else(|l| l);
+			pool.insert(location, item);
+
+			<Pool<T, I>>::put(&pool);
+			Self::deposit_event(RawEvent::CandidateScored);
+		}
+
+		/// Root-dispatchable call to change `MemberCount`.
+		pub fn change_member_count(origin, count: u32) {
+			ensure_root(origin)?;
+			<MemberCount<I>>::put(&count);
+		}
+	}
+}
+
+impl<T: Trait<I>, I: Instance> Module<T, I> {
+
+	/// Fetches the `MemberCount` highest scoring members from
+	/// `Pool` and puts them into `Members`.
+	///
+	/// If `notify` is set to true `T::MembershipChanged::change_members_sorted`
+	/// will be invoked at the end of the method.
+	fn refresh_members(notify: bool) {
```

Comment:
why would notify ever be `false`?

Reply:
This was used to differentiate between calling `T::MembershipChanged` vs. `T::MembershipInitialized`. I've refactored the code now to make this clearer.